tensorboard/plugins/pr_curve/summary.py

```diff
@@ -0,0 +1,190 @@
+
+    with tf.name_scope('variables'):
+      # Now create the variables which correspond to the bucket values.
+      # These are flat arrays with num_thresholds valuewchargins per class.
+      tp_buckets_v = tf.Variable(
+          tf.zeros([num_classes * num_thresholds], dtype=dtype),
+          name='tp_buckets',
+          trainable=False,
+          collections=[tf.GraphKeys.LOCAL_VARIABLES])
+      fp_buckets_v = tf.Variable(
+          tf.zeros([num_classes * num_thresholds], dtype=dtype),
+          name='fp_buckets',
+          trainable=False,
+          collections=[tf.GraphKeys.LOCAL_VARIABLES])
+
+    initialize_bucket_counts = tf.variables_initializer(
+        [tp_buckets_v, fp_buckets_v])
+    with tf.control_dependencies([initialize_bucket_counts]):
+      # Create the non-flat views with class_index as first dimension.
+      tp_buckets = tf.reshape(tp_buckets_v, (num_classes, -1))
+      fp_buckets = tf.reshape(fp_buckets_v, (num_classes, -1))
+
+      with tf.name_scope('update_op'):
+        # Use scatter_add to update the buckets.
+        update_tp = tf.scatter_add(
+            tp_buckets_v, bucket_indices, true_labels, use_locking=True)
+        update_fp = tf.scatter_add(
+            fp_buckets_v, bucket_indices, false_labels, use_locking=True)
+
+    with tf.control_dependencies([update_tp, update_fp]):
+      with tf.name_scope('metrics'):
+        thresholds = tf.cast(
+            tf.linspace(0.0, 1.0, num_thresholds), dtype=dtype)
+        # Set up the cumulative sums to compute the actual metrics.
+        tp = tf.cumsum(tp_buckets, reverse=True, axis=1, name='tp')
+        fp = tf.cumsum(fp_buckets, reverse=True, axis=1, name='fp')
+        # fn = sum(true_labels) - tp
+        #    = sum(tp_buckets) - tp
+        #    = tp[:, 0] - tp
+        # Similarly,
+        # tn = fp[:, 0] - fp
+        tn = tf.subtract(fp[:, 0:1], fp, name='tn')
+        fn = tf.subtract(tp[:, 0:1], tp, name='fn')
+
+        # Store the number of thresholds within the summary metadata because
+        # that value is constant for all pr curve summaries with the same tag.
+        summary_metadata = tf.SummaryMetadata()
+        pr_curve_plugin_data = pr_curve_pb2.PrCurvePluginData(
+            num_thresholds=num_thresholds)
+        summary_metadata.plugin_data.add(
+            plugin_name='pr_curve',
+            content=json_format.MessageToJson(pr_curve_plugin_data))
+
+        # Store counts within a tensor. We store them in the order:
+        # true positives, false positives, true negatives, and false negatives.
+        combined_counts = tf.concat([tp, fp, tn, fn], axis=0)
+        reshaped_counts = tf.reshape(
+            combined_counts, (4, num_classes, num_thresholds))
+
+    return tf.summary.tensor_summary(
+        name=tag,
+        tensor=tf.cast(reshaped_counts, tf.int64),
+        summary_metadata=summary_metadata,
+        display_name=display_name if display_name is not None else tag)
```

Comment:
OK - now we pass `display_name` and `summary_description` into the constructor of the `SummaryMetadata` proto. `display_name` used to be passed into the `tensor_summary` method.